src/components/Cluster/ClusterDetail/V5ClusterDetailTable.js

```diff
@@ -306,37 +306,45 @@ class V5ClusterDetailTable extends React.Component {
     }
   }
 
-  // TODO Move this to the action creator so it will be triggered on every NPs load.
   produceNodePools = () => {
-    if (Object.keys(this.props.nodePools).length > 0) {
-      this.setState({ loading: true });
-      const nodePools = clusterNodePools(
-        this.props.nodePools,
-        this.props.cluster
-      );
-
-      const allZones = nodePools
-        ? nodePools
-            .reduce((accumulator, current) => {
-              return [...accumulator, ...current.availability_zones];
-            }, [])
-            .map(zone => zone.slice(-1))
-        : [];
-
-      // This array stores available zones that are in at least one node pool.
-      // We only want unique values because this is used fot building the grid.
-      const availableZonesGridTemplateAreas = [...new Set(allZones)]
-        .sort()
-        .join(' ');
+    // If there are no node pools or after all node pools are removed, set local
+    // state to an empty array
+    if (Object.keys(this.props.nodePools).length === 0) {
       this.setState({
-        availableZonesGridTemplateAreas: `"${availableZonesGridTemplateAreas}"`,
-      });
-
-      this.setState({
-        nodePools,
+        nodePools: [],
```

Comment:
Hey Oriol! This is something I just noticed, I know it's a comment on something that existed already before this PR, so whatever comes out of it might not be in scope of these changes... but:

Why is nodepools in the state? Calling this produceNodePools in componentDidMount feels a bit weird to me.

I think this is a different pattern than what we normally do right?

Is there a way to move this code to mapStateToProps?

Or is the actual work being done here so 'heavy' that we want to do it after the component mounts so we can show a loading spinner for just that?

Reply:
Yes, you are totally right, this shouldn't be here. And no, this is not an expensive computation. 

This `produceNodePools` function does two things:

1. It selects cluster node pools 
2. It generates a AZs array for each NP

What I want to do is put 1 in `mapStateToProps`, using a selector (that we already have in `clustersSelectors`) and substitute `state.nodePools` using instead of it  something like `state.nodePoolsAZs`

I just haven't had the time to do it.

Reply:
Ok cool! No worries! Thanks for explaining :)

Reply:
Thanks for noticing it! I love to discuss this kind of things :-)